crates/volta-migrate/src/v2.rs

```diff
@@ -94,24 +96,23 @@ impl TryFrom<V1> for V2 {
 /// This will ensure that we don't treat the default npm from a prior version of Volta as a "custom" npm that
 /// the user explicitly requested
 fn clear_default_npm(platform_file: &Path) -> Fallible<()> {
-    if platform_file.exists() {
-        let platform_json =
-            read_to_string(platform_file).with_context(|| ErrorKind::ReadPlatformError {
-                file: platform_file.to_owned(),
-            })?;
-        let mut existing_platform = Platform::from_json(platform_json)?;
-
-        if let Some(ref mut node_version) = &mut existing_platform.node {
-            if let Some(npm) = &node_version.npm {
-                if let Ok(default_npm) = load_default_npm_version(&node_version.runtime) {
-                    if *npm == default_npm {
-                        node_version.npm = None;
-                        write(platform_file, existing_platform.into_json()?).with_context(
-                            || ErrorKind::WritePlatformError {
-                                file: platform_file.to_owned(),
-                            },
-                        )?;
-                    }
+    let platform_json = read_to_string(platform_file)
+        .or_else(ok_if_not_found)
+        .with_context(|| ErrorKind::ReadPlatformError {
+            file: platform_file.to_owned(),
+        })?;
```

Comment:
This happens to work because the following line `Platform::from_json` handles an empty string, but I'm not sure if we should be relying on that behavior. Instead, I think this can be a little more explicit about handling the `NotFound` error (similar to the implementation of `from_file_if_exists` above):

```rust
let platform_json = match read_to_string(platform_file) {
    Ok(json) => json,
    Err(error) => if error.kind() == io::ErrorKind::NotFound {
        return Ok(());
    } else {
        return Err(VoltaError::from_source(error, ErrorKind::ReadPlatformError { ... }));
    }
};
```